cmd/cluster/owner.go

```diff
@@ -0,0 +1,160 @@
+package cluster
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"os"
+	"strings"
+
+	"github.com/openshift-online/ocm-cli/pkg/ocm"
+	sdk "github.com/openshift-online/ocm-sdk-go"
+	v1 "github.com/openshift-online/ocm-sdk-go/accountsmgmt/v1"
+	"github.com/openshift/osdctl/internal/utils/globalflags"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+// ownerOptions defines the struct for the current command
+// This command requires the ocm API Token https://cloud.redhat.com/openshift/token be available in the OCM_TOKEN env variable.
+type ownerOptions struct {
+	output   string
+	verbose  bool
+	userName string
+
+	genericclioptions.IOStreams
+	GlobalOptions *globalflags.GlobalOptions
+}
+
+// newCmdOwner return a new command
+func newCmdOwner(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags, globalOpts *globalflags.GlobalOptions) *cobra.Command {
+	ops := newOwnerOptions(streams, flags, globalOpts)
+	ownerCmd := &cobra.Command{
+		Use:               "owner",
+		Short:             "List the clusters owned by the user (can be specified to any user, not only yourself)",
+		Args:              cobra.NoArgs,
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
+		},
+	}
+	ownerCmd.Flags().StringVarP(&ops.userName, "user-id", "u", ops.userName, "user to check the cluster owner on")
+
+	return ownerCmd
+}
+
+func newOwnerOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags, globalOpts *globalflags.GlobalOptions) *ownerOptions {
+	return &ownerOptions{
+		IOStreams:     streams,
+		GlobalOptions: globalOpts,
+	}
+}
+
+func (o *ownerOptions) complete(cmd *cobra.Command, _ []string) error {
+
+	o.output = o.GlobalOptions.Output
+
+	return nil
+}
+
+func createConnection() (*sdk.Connection, error) {
+	connection, err := ocm.NewConnection().Build()
+	if err != nil {
+		// TODO: this should be tracked by bubbling an error up and, exposing so we can compare it here.
+		if strings.Contains(err.Error(), "Not logged in, run the") {
```

Comment:
Is there an error code we can check here instead? Also we can combine 68 with 70, start with a base error string and append "please run ocm login" to the message if the error code is hit

Reply:
no, they just return a `fmt.Errorf` with the message.

Reply:
K, let's do that then :/

Reply:
This should be better.. TAL?